src/library/rssscanner.c

```diff
@@ -0,0 +1,730 @@
+    }
+  else
+    {
+      /* expect json resp - get feedUrl
+       * {
+       *  "resultCount": 1,
+       *  "results": [
+       *    {
+       *      "wrapperType": "track",
+       *      "kind": "podcast",
+       *      ...
+       *      "collectionViewUrl": "https://podcasts.apple.com/us/podcast/cgp-grey/id974722423?uo=4",
+       *      "feedUrl": "http://cgpgrey.libsyn.com/rss",
+       *      ...
+       *      "genres": [
+       *        "Education",
+       *        "Podcasts",
+       *        "News"
+       *      ]
+       *    }
+       *  ]
+       *}
+       */
+      if (json_object_object_get_ex(json, "results", &jsonra) && (feedURL = jparse_str_from_array(jsonra, 0, "feedUrl")) )
+        {
+          buf = strcpy(malloc(strlen(feedURL)+1), feedURL);
+	  DPRINTF(E_DBG, L_LIB, "mapped apple podcast URL: %s -> %s\n", rss_url, buf);
+        }
+      else
+        DPRINTF(E_DBG, L_LIB, "Could not parse feedURL from RSS apple, podcast id %u\n", podid);
+    }
+
+  jparse_free(json);
+  evbuffer_free(evbuf);
+  return buf;
+}
+
+#ifdef RSS_DEBUG
+static void
+rss_playlist_items(int plid)
+{
+  struct query_params qp;
+  struct db_media_file_info dbpli;
+  int ret;
+
+  memset(&qp, 0, sizeof(struct query_params));
+
+  qp.type = Q_PLITEMS;
+  qp.idx_type = I_NONE;
+  qp.id = plid;
+
+  ret = db_query_start(&qp);
+  if (ret < 0)
+    {
+      db_query_end(&qp);
+      return;
+    }
+  while (((ret = db_query_fetch_file(&qp, &dbpli)) == 0) && (dbpli.id))
+    {
+      DPRINTF(E_LOG, L_LIB, "plid=%u  { id=%s title=%s path=%s }\n", plid, dbpli.id, dbpli.title, dbpli.path);
+    }
+  db_query_end(&qp);
+
+  return;
+}
+#endif
+
+// only add required number of feeds items when limit > 0
+int
+rss_scan_feed(int pl_id, const char *url, long limit, unsigned *nadded)
+{
+  struct media_file_info mfi;
+  char *vpath = NULL;
+  int feed_file_id;
+  struct tm tm;
+  unsigned vpathlen = 0;
+  unsigned len = 0;
+  time_t mtime;
+
+  char *apple_url = NULL;
+
+  const char *rss_xml = NULL;
+  mxml_node_t *tree = NULL;
+  mxml_node_t *channel;
+  mxml_node_t *node;
+  mxml_node_t *item;
+  const char *rss_feed_title = NULL;
+  const char *rss_feed_author = NULL;
+  const char *rss_item_title = NULL;
+  const char *rss_item_pubDate = NULL;
+  const char *rss_item_url = NULL;
+  const char *rss_item_link = NULL;
+  const char *rss_item_type = NULL;
+
+  struct http_client_ctx ctx;
+  struct evbuffer *evbuf;
+
+  int ret = -1;
+
+  DPRINTF(E_DBG, L_LIB, "Refreshing RSS id: %u url: %s limit: %ld\n", pl_id, url, limit);
+  db_pl_ping(pl_id);
+  db_pl_ping_items_bymatch("http://", pl_id);
+  db_pl_ping_items_bymatch("https://", pl_id);
+
+  evbuf = evbuffer_new();
+  if (!evbuf)
+    goto cleanup;
+
+  // Is it an apple podcast stream?
+  // ie https://podcasts.apple.com/is/podcast/cgp-grey/id974722423
+  if (strncmp(url, "https://podcasts.apple.com/", 27) == 0)
+    apple_url = process_apple_rss(url);
```

Comment:
I've seen more and more of my podcasts appear advertised more heavily on Apple even if they have their own dedicated websites; the BBC have this.  But ultimately the Apple podcast id just points back at the content publsher's own RSS feed anyway.

To that end, and the rising popularity of apple podcast addresses, there is a convience function here to translate the apple podcast links to the underlying the RSS so we can actually parse the XML.  Note we store the apple podcast url (not the translated) so we will perform translation every time we do our periodic refresh which ensures that if content creator moves their RSS feed to another site, the server won't lose updates to their podcast.